var/ramble/repos/builtin/applications/minixyce/application.py

```diff
@@ -0,0 +1,145 @@
+                variants='+legacylaunchers +pmi +cxx')
+
+    software_spec('minixyce',
+                  base='minixyce',
+                  version='1.0',
+                  compiler='gcc12',
+                  variants='+mpi',
+                  mpi='ompi415',
+                  required=True)
+
+    executable('execute', 'miniXyce.x --circuit {workload_name}.net --pf params.txt', use_mpi=True)
+
+    executable('get_simple_network',
+               template=['cp {minixyce}/doc/tests/{workload_name}.net {experiment_run_dir}/{workload_name}.net'],
+               use_mpi=False)
+
+    executable('generate_RC_ladder',
+               template=['perl {minixyce}/doc/tests/RC_ladder.pl {num_ladder_stages} > {experiment_run_dir}/{workload_name}.net; echo Running perl'],
+               use_mpi=False)
+
+    executable('generate_RLC_ladder',
+               template=['perl {minixyce}/doc/tests/RLC_ladder.pl {num_ladder_stages} > {experiment_run_dir}/{workload_name}.net; echo Running perl'],
+               use_mpi=False)
+
+    executable('generate_RC_ladder2',
+               template=['perl {minixyce}/doc/tests/RC_ladder2.pl {num_ladder_stages} > {experiment_run_dir}/{workload_name}.net; echo Running perl'],
+               use_mpi=False)
+
+    executable('generate_RLC_ladder2',
+               template=['perl {minixyce}/doc/tests/RLC_ladder2.pl {num_ladder_stages} > {experiment_run_dir}/{workload_name}.net; echo Running perl'],
+               use_mpi=False)
+
+    cir_workloads = ['cir1', 'cir2', 'cir3', 'cir4', 'cir5']
+    for cir_workload in cir_workloads:
+        workload(cir_workload, executables=['get_simple_network', 'execute'])
+
+    rc_workloads = ['RC_ladder', 'RLC_ladder', 'RC_ladder2', 'RLC_ladder2']
+    for workload_name in rc_workloads:
+        workload(workload_name, executables=['generate_' + workload_name, 'execute'])
+
+    all_workloads = cir_workloads + rc_workloads
+    workload_variable('t_start', default='0', description='Start time', workloads=all_workloads)
+    workload_variable('t_step', default='1e-8', description='Time step', workloads=all_workloads)
+    workload_variable('t_stop', default='5e-6', description='Stop time', workloads=all_workloads)
+    workload_variable('tol', default='1e-6', description='Tolerance', workloads=all_workloads)
+    workload_variable('k', default='10', description='Tells GMRES how often to restart', workloads=all_workloads)
+
+    workload_variable('num_ladder_stages', default='8',
+                      description='Number of Ladder stages for RC|RLC ladder|ladder2 test generation',
+                      workloads=rc_workloads)
+
+    log_file = '{experiment_run_dir}/{workload_name}_tran_results.prn'
+    processed_output = '{experiment_run_dir}/processed_output.txt'
+
+    result_regex = r'.*\s+(?P<num_iters>[0-9]+)\s+(?P<num_restarts>[0-9]+)'
+
+    floating_point_regex = r'[\+\-]*[0-9]*\.*[0-9]+E*[\+\-]*[0-9]*'
+    close_paren = r')'
```

Comment:
I'm not sure if `close_paren` adds much. It's usage isn't consistent here, and the actually string is shorter than the variable name.

I'm not opposed to the idea of using the variable instead of the string, but I don't think we should mix and match the semantics in this file.

Reply:
success_criteria:

Sample tests cir1 through cir4 can be run and then checked using a provided perl script. This simply matches exact numeric results.

In general, changing platforms should have the same final FoM values on any platform.  Should we provide comparisons to be checked for certain workloads?

Reply:
I'm curious what this would look like. Is it just running a perl script, taking the expected output, and hard-coding it into the python file?

If so, that seems like a reasonable thing to try. Though we might want to test that the values don't change on a handful of platforms before we commit to something like that.

Reply:
Added success_criteria that verifies the job ran.